main.js

```diff
@@ -114,6 +114,20 @@ updateNotifier({pkg: cli.pkg}).notify();
 
 const {input, flags: opts} = cli;
 
+// Make data types for `opts.space` match those of the API
+if (typeof opts.space === 'string') {
+	if (/^\d$/.test(opts.space)) {
+		opts.space = parseInt(opts.space, 10);
+	} else if (opts.space) {
+		// Assume `opts.space` was set to a filename when run as `xo --space file.js`
+		input.push(opts.space);
+	}
+
+	if (typeof opts.space !== 'number') {
```

Comment:
I'm happy to adjust that. Here's my reasoning for doing it this way: in the non-number branch of the first `if` statement, I wouldn't want to unconditionally append to `input` since the value could be `''`, which wouldn't be useful as a file name. I could absolutely nest this test inside of the first one, though.

Reply:
It makes sense now that you explain it, but by reading the code the fact you meant to unconditional append to `input `since the value could be `''` is really not clear. It not really clear that the value can be `''` unless you know all details of the parser by heart.
The code is locally valid, but most likely if we try to modify it in 6 months we'll never figure that out.

Reply:
Good point! When I move that condition into the first block I can also add a comment to that effect.